src/systems/userinput/devices/app-aware-touchscreen.js

```diff
@@ -149,44 +161,51 @@ export class AppAwareTouchscreenDevice {
       return;
     }
 
-    if (!jobIsAssigned(MOVE_CURSOR_JOB, this.assignments) && shouldMoveCursor(touch, this.raycaster)) {
-      const assignment = assign(touch, MOVE_CURSOR_JOB, this.assignments);
+    if (this.assignments.length === 0) {
+      let assignment;
+
+      // First touch
+      if (shouldMoveCursor(touch, this.raycaster)) {
+        assignment = assign(touch, MOVE_CURSOR_JOB, this.assignments);
+
+        // Grabbing objects is delayed by several frames:
+        // - We don't want the physics constraint to be applied too early, which results in crazy velocities
+        // - We don't want to mis-trigger grabs if the user is about to put down a second finger.
+        assignment.framesUntilGrab = 2;
```

Comment:
This is a scary hack but I don't have a better idea for how to fix things. Glad it works. I expect that we're going to have to revisit this as the app code changes. (It may be affected if we change physics, cursor-controller, or tick order.)

Reply:
yep in the end this was a cleaner way to deal the grab for the pinch case, so i think this will stick around even if we dropped the need to delay for the physics issue.